### PR TITLE
Release 0.10.0

### LINKED ISSUE #280: Release 0.10.0
@adam-becker, @TysonMN, we should be doing a release after merging

* <del>https://github.com/hedgehogqa/fsharp-hedgehog/pull/238</del> *Postponed for 0.11.0 or 1.0.00*
* https://github.com/hedgehogqa/fsharp-hedgehog/pull/247
* https://github.com/hedgehogqa/fsharp-hedgehog/pull/275

I think those are ready to merge, or close to be ready to merge. This way we can get the Hedgehog.Linq stuff out soon. And then focus on the other ones.

### PR DESCRIPTION
**Pull request by moodmosaic**
https://github.com/hedgehogqa/fsharp-hedgehog/compare/v0.9.0...0bbc3fda6b8da4ac39a28c6ea21e43c007e27250

fixes #280

/cc @adam-becker @TysonMN @dharmaturtle @ThisFunctionalTom @cmeeren

:tada:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+- Add `OptionTree.traverse` ([#282][282], [@TysonMN][TysonMN])
+- Use fsdocs ([#277][277], [@adam-becker][adam-becker])
+- Rearrange parameters for better chaining ([#266][266], [@adam-becker][adam-becker])
+  - Rearrange `Tree.bind` parameters ([#300][300], [@adam-becker][adam-becker])
+- Add `ListGen.traverse` ([#260][260], [@TysonMN][TysonMN])
+- Improve and extend DateTime/DateTimeOffset generation ([#252][252], [@cmeeren][cmeeren])
+- Split Property.fs across multiple files ([#247][247], [@adam-becker][adam-becker])
+- Add support for LINQ via Hedgehog.Linq namespace ([#244][244], [@adam-becker][adam-becker])
+- Switch to Fable.Mocha ([#196][196], [@ThisFunctionalTom][ThisFunctionalTom])
+
```
Comment (moodmosaic) [inline CHANGELOG.md:13]: @adam-becker, here in the CHANGELOG.md I'm including the most high-level stuff.

File: CHANGELOG.md [hunk 2]
```diff
@@ -117,10 +132,30 @@
...
 [255]:
   https://github.com/hedgehogqa/fsharp-hedgehog/pull/255
+[252]:
+  https://github.com/hedgehogqa/fsharp-hedgehog/pull/252
 [250]:
   https://github.com/hedgehogqa/fsharp-hedgehog/pull/250
+[247]:
+  https://github.com/hedgehogqa/fsharp-hedgehog/pull/247
+[244]:
+  https://github.com/hedgehogqa/fsharp-hedgehog/pull/244
```
Comment (moodmosaic) [inline CHANGELOG.md:158]: @adam-becker, here I'm adding each one of the PR URLs from above. So far this is done manually.

File: src/Hedgehog/Hedgehog.fsproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>netstandard1.6;netstandard2.0;net45</TargetFrameworks>
-    <Version>0.9.0</Version>
+    <Version>0.10.0</Version>
```
Comment (moodmosaic) [inline src/Hedgehog/Hedgehog.fsproj:5]: @adam-becker, the most important change, of course, is this one.

**Comment by moodmosaic** [comment]
What's next @TysonMN (according to #215)? :eyes:

**Comment by TysonMN** [comment]
I think just completing the PR.  If I recall correctly, that automation looks at the project file and publishes a NuGet package if the value of the `Version` element has changed.

**Comment by moodmosaic** [comment]
First we need to push a tag, then. I can do it during the weekend.

**Comment by TysonMN** [comment]
I think this automated process also adds a tag that matches the value of the new `Version` element.

**Comment by dharmaturtle** [comment]
`[brandedoutcast/publish-nuget](https://github.com/brandedoutcast/publish-nuget)`, which we use, will automatically make the tag. We didn't flick the `TAG_COMMIT` and it's on by default. I believe the tag format is good.

**Comment by TysonMN** [comment]
(lol, beat you by five seconds!)

File: CHANGELOG.md (hunk 2 above)
Comment (adam-becker) [reply CHANGELOG.md:158]: @moodmosaic Hopefully putting the issues into a milestone makes this part of the release easier.

File: src/Hedgehog/Hedgehog.fsproj (hunk 3 above)
Comment (adam-becker) [reply src/Hedgehog/Hedgehog.fsproj:5]: @moodmosaic At work, we've been using [GitVersion][gitversion] to grab version information from git, it works pretty nicely in my opinion.

[gitversion]: https://gitversion.net/docs/

**Comment by moodmosaic** [comment]
Look like it worked 👍 https://www.nuget.org/packages/Hedgehog/0.10.0

**Comment by moodmosaic** [comment]
Also the tag was created 👍 

@adam-becker, I had to create the release manually (from the tag). Also, I need to edit the release so it has some release notes inside. I'll get back to this during the weekend.

**Comment by adam-becker** [comment]
@moodmosaic I generated the new documentation pages as well. If a tag were pushed as part of the release we could automate the documentation generation part, the only input to the script is the version number.

**Comment by moodmosaic** [comment]
@adam-becker, yes, I'm all in for including more stuff in the release pipeline, so there's less manual steps to be done afterwards. It's not like we're releasing everyday, so even if the stuff we add makes the CI build slow(er) I think it's okay.

**Comment by moodmosaic** [comment]
> @adam-becker, I had to create the release manually (from the tag). Also, I need to edit the release so it has some release notes inside. I'll get back to this during the weekend.

I've now just edited the v0.10.0 release notes (as per the CHANGELOG.md) https://github.com/hedgehogqa/fsharp-hedgehog/releases/tag/v0.10.0

**Pull request merged**